exist-core/src/main/java/org/exist/util/Configuration.java

```diff
@@ -325,272 +346,290 @@ private void configureLockManager(final Element lockManager) throws DatabaseConf
+        LOG.debug(XQueryContext.PROPERTY_XQUERY_RAISE_ERROR_ON_FAILED_RETRIEVAL + ": {}",
+                config.get(XQueryContext.PROPERTY_XQUERY_RAISE_ERROR_ON_FAILED_RETRIEVAL));
 
-        final String trace = getConfigAttributeValue( xquery, PerformanceStats.CONFIG_ATTR_TRACE );
-        config.put( PerformanceStats.CONFIG_PROPERTY_TRACE, trace );
+        final String trace = getConfigAttributeValue(xquery, PerformanceStats.CONFIG_ATTR_TRACE);
+        config.put(PerformanceStats.CONFIG_PROPERTY_TRACE, trace);
 
         // built-in-modules
-        final Map<String, Class<?>> classMap      = new HashMap<>();
-        final Map<String, String>   knownMappings = new HashMap<>();
-        final Map<String, Map<String, List<? extends Object>>> moduleParameters = new HashMap<>();
+        final Map<String, Class<?>> classMap = new HashMap<>();
+        final Map<String, String> knownMappings = new HashMap<>();
+        final Map<String, Map<String, List<?>>> moduleParameters = new HashMap<>();
         loadModuleClasses(xquery, classMap, knownMappings, moduleParameters);
-        config.put( XQueryContext.PROPERTY_BUILT_IN_MODULES, classMap);
-        config.put( XQueryContext.PROPERTY_STATIC_MODULE_MAP, knownMappings);
-        config.put( XQueryContext.PROPERTY_MODULE_PARAMETERS, moduleParameters);
+        config.put(XQueryContext.PROPERTY_BUILT_IN_MODULES, classMap);
+        config.put(XQueryContext.PROPERTY_STATIC_MODULE_MAP, knownMappings);
+        config.put(XQueryContext.PROPERTY_MODULE_PARAMETERS, moduleParameters);
     }
 
     /**
      * Read list of built-in modules from the configuration. This method will only make sure
      * that the specified module class exists and is a subclass of {@link org.exist.xquery.Module}.
      *
-     * @param   xquery            configuration root
-     * @param   modulesClassMap   map containing all classes of modules
-     * @param   modulesSourceMap  map containing all source uris to external resources
-     *
-     * @throws  DatabaseConfigurationException
+     * @param xquery           configuration root
+     * @param modulesClassMap  map containing all classes of modules
+     * @param modulesSourceMap map containing all source uris to external resources
+     * @throws DatabaseConfigurationException if one of the modules is configured incorrectly
      */
-    private void loadModuleClasses( Element xquery, Map<String, Class<?>> modulesClassMap, Map<String, String> modulesSourceMap, Map<String, Map<String, List<? extends Object>>> moduleParameters) throws DatabaseConfigurationException {
+    private void loadModuleClasses(final Element xquery,
+                                   final Map<String, Class<?>> modulesClassMap,
+                                   final Map<String, String> modulesSourceMap,
+                                   final Map<String, Map<String, List<?>>> moduleParameters
+    ) throws DatabaseConfigurationException {
```

Comment:
Changed from `non-reified` to `reified` types. Is that intentional?

Reply:
Same over-eager optimisation as in #4838
I am gonna change that.

Reply:
done